addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NikoHomeControlCommunication.java

```diff
@@ -0,0 +1,487 @@
+        }
+    }
+
+    private void cmdListLocations(List<HashMap<String, String>> data) {
+
+        logger.debug("Niko Home Control: list locations");
+
+        this.locations.clear();
+
+        for (HashMap<String, String> location : data) {
+            int id = Integer.valueOf(location.get("id"));
+            String name = location.get("name");
+            NhcLocation nhcLocation = new NhcLocation(name);
+            this.locations.put(id, nhcLocation);
+        }
+    }
+
+    private void cmdListActions(List<HashMap<String, String>> data) {
+
+        logger.debug("Niko Home Control: list actions");
+
+        for (HashMap<String, String> action : data) {
+
+            int id = Integer.valueOf(action.get("id"));
+            Integer state = Integer.valueOf(action.get("value1"));
+
+            if (!this.actions.containsKey(id)) {
+                // Initial instantiation of NhcAction class for action object
+                String name = action.get("name");
+                Integer type = Integer.valueOf(action.get("type"));
+                Integer locationId = Integer.valueOf(action.get("location"));
+                String location = null;
+                if (locationId != null) {
+                    location = this.locations.get(locationId).getName();
+                }
+                NhcAction nhcAction = new NhcAction(id, name, type, location);
+                nhcAction.setState(state);
+                nhcAction.setNhcComm(this);
+                this.actions.put(id, nhcAction);
+            } else {
+                // Action object already exists, so only update state.
+                // If we would re-instantiate action, we would lose pointer back from action to thing handler that was
+                // set in thing handler initialize().
+                this.actions.get(id).setState(state);
+            }
+        }
+    }
+
+    private void cmdExecuteActions(Map<String, String> data) {
+
+        Integer errorCode = Integer.valueOf(data.get("error"));
+        if (errorCode.equals(0)) {
+            logger.debug("Niko Home Control: execute action success");
+        } else {
+            logger.warn("Niko Home Control: error code {} returned on command execution", errorCode);
+        }
+    }
+
+    private void eventListActions(List<HashMap<String, String>> data) {
+
+        for (HashMap<String, String> action : data) {
+            int id = Integer.valueOf(action.get("id"));
+            if (!this.actions.containsKey(id)) {
+                logger.warn("Niko Home Control: action in controller not known to openHab {}", id);
+                return;
+            }
+            Integer state = Integer.valueOf(action.get("value1"));
+            logger.debug("Niko Home Control: event execute action {} with state {}", id, state);
+            try {
+                this.actions.get(id).setState(state);
+            } catch (Exception e) {
```

Comment:
No, I don't need to. Something is very wrong when I would get here. It would be a programming error. So I removed the try/catch.